node/lib/util/git_util.js

```diff
@@ -128,6 +128,27 @@ exports.isValidRemoteName = co.wrap(function *(repo, name) {
 });
 
 /**
+ * Return the URL for the remote named "origin" in the specified `repo`, or
+ * null if there is no remote named "origin".
```

Comment:
They mostly end up passing the result to `SubmoduleConfigUtil.resolveSubmoduleUrl`, which will throw a `UserError` only if the submodule URL is relative. I wanted to have this behavior so that everything works in the case that the meta-repo doesn't have an origin, but no relative URLs are used -- frequently in testing.